### DISCUSSION TITLE
Who is msg.sender when an account calls a function in Contract A which in turn calls a library function that incorporates msg.sender?

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#864] [General]
Who is msg.sender when an account calls a function in Contract A which in turn calls a library function that incorporates msg.sender?....is it contract A or is it the account that called the function in contract A?

Does it make any difference if you are using a diamond interface (proxy contract) for this set of contracts?

**Comment by 0xusmanf** [comment]
> Who is msg.sender when an account calls a function in Contract A which in turn calls a library function that incorporates msg.sender?....is it contract A or is it the account that called the function in contract A?

If contract A is calling the library function with other than `delegatecall` then `msg.sender` will be contract A.
If contract A is calling the library function with `delegatecall` then `msg.sender` will be the account that called the function in contract A.

> Does it make any difference if you are using a diamond interface (proxy contract) for this set of contracts?

Can't say much about it without having more context.
